src/message.c

```diff
@@ -608,33 +777,53 @@ int message_set_application_properties(MESSAGE_HANDLE message, AMQP_VALUE applic
 {
 	int result;
 
-	if ((message == NULL) ||
-		(application_properties == NULL))
+	if (message == NULL)
 	{
-		LogError("Bad arguments: message = %p, application_properties = %p",
-			message, application_properties);
+		/* Codes_SRS_MESSAGE_01_066: [ If `message` is NULL, `message_set_application_properties` shall fail and return a non-zero value. ]*/
+		LogError("NULL message");
 		result = __FAILURE__;
 	}
 	else
 	{
 		AMQP_VALUE new_application_properties;
 
-		new_application_properties = application_properties_clone(application_properties);
-		if (new_application_properties == NULL)
-		{
-			LogError("Cannot clone application properties");
-			result = __FAILURE__;
-		}
-		else
+		/* Tests_SRS_MESSAGE_01_149: [ If `application_properties` is NULL, the previously stored application properties associated with `message` shall be freed. ]*/
+		if (application_properties == NULL)
 		{
 			if (message->application_properties != NULL)
 			{
 				amqpvalue_destroy(message->application_properties);
```

Comment:
Done.

---
In reply to: [120404111](https://github.com/Azure/azure-uamqp-c/pull/128#discussion_r120404111) [](ancestors = 120404111)